bson-numpy/bsonnumpy.c

```diff
@@ -428,45 +341,36 @@ _load_array_from_bson(bson_iter_t *bsonit, PyArrayObject *ndarray,
 
     /* Load data into array */
     bson_iter_recurse(bsonit, &sub_it);
-    if (count == 1) {
```

Comment:
This is a substantial change. I was testing sequence_to_numpy and it suddenly didn't make sense to me anymore that we interpret BSON arrays of length 1 as scalars. If this ever arises in real life, I think we should treat this as a real dimension of size 1, not ignore it.